lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1546,6 +1546,30 @@ def test_add_cell_measure(self):
         cube.add_cell_measure(a_cell_measure, [0, 1])
         self.assertEqual(cube.cell_measure('area'), a_cell_measure)
 
+    def test_add_valid_aux_factory(self):
+        cube = Cube(np.arange(8).reshape(2, 2, 2))
+        delta = AuxCoord(points=[0, 1], long_name='delta', units='m')
+        sigma = AuxCoord(points=[0, 1], long_name='sigma')
+        orog = AuxCoord(np.arange(4).reshape(2, 2), units='m')
+        cube.add_aux_coord(delta, 0)
+        cube.add_aux_coord(sigma, 0)
+        cube.add_aux_coord(orog, (1, 2))
+        factory = HybridHeightFactory(delta=delta, sigma=sigma, orography=orog)
+        self.assertIsNone(cube.add_aux_factory(factory))
```

Comment:
Sorry @hdyson, I'm a bit confuse about what this test is for.  Can you maybe add a comment or two to explain please?  I'm probably being slow but I don't really understand what we are asserting here.

Reply:
I didn't see an existing test for adding a valid aux factory.  Since I wanted to test for a particular error with an invalid aux factory, it felt like a good idea to include that test to confirm I wasn't breaking existing behaviour.  There are tests for adding dim coords, aux coords and cell measures, I think the lack of an existing test for an aux factory may be an oversight?

Reply:
You're quite right.  I often despair at the testing (or lack of), but let's not get into that.  Adding tests that should already be there is great.

I'm still not quite sure what the `assertIsNone` asserts there is none of though.  That last line is slightly information dense for my delicate and easily-confused taste.

Reply:
Ah, okay, now I understand.  I'm using `assertIsNone` as an analogue for `assertDoesNotRaise` (which doesn't exist).  Would a comment help, or just removing the assertion?  Personally, it always makes me nervous when I see a test with no assert in it, it always makes me wonder if it's actually been finished or not.

Reply:
I think just a comment explaining the assert (as you just have to me, thanks for that) is enough.